File: 739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        int n = temp.size(), ind = -1;
        int* s = (int*)malloc(sizeof(int) * n);
        vector<int> res(n);
        for (int i = n - 1; i >= 0; --i) {
            while (ind >= 0 && temp[i] >= temp[s[ind]])
                --ind;
            res[i] = (ind < 0) ? 0 : (s[ind] - i);
            s[++ind] = i;
        }
        free(s);
        return res;
    }
};

File: 739.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temp) {
    let n = temp.length, ind = -1;
    let s = []
    let res = []
    for (let i = n - 1; i >= 0; --i) {
        while (ind >= 0 && temp[i] >= temp[s[ind]])
            --ind;
        if (ind < 0)
            res[i] = 0;
        else
            res[i] = s[ind] - i;
        s[++ind] = i;
    }
    return res;
};
